com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaBuilder.java

```diff
@@ -253,25 +274,36 @@ public void build(Task<Void> task) throws CompileExceptionError, IOException {
             }
         }
 
-        // if for some reason, the project needs to be bundled with regular Lua files (e.g. on iOS)
-        boolean use_vanilla_lua = this.project.option("use-vanilla-lua", "false").equals("true");
-        if (use_vanilla_lua) {
-            for(IResource res : task.getOutputs()) {
-                String path = res.getAbsPath();
-                if(path.endsWith("luac") || path.endsWith("scriptc") || path.endsWith("gui_scriptc") || path.endsWith("render_scriptc")) {
+        boolean use_lua_source = this.project.option("use-lua-source", "false").equals("true");
+        
+        // set compression and encryption flags
+        // if the use-lua-source flag is set the project will use uncompressed plain text Lua script files
+        // if the use-lua-source flag is NOT set the project will use encrypted and possibly also compressed bytecode
+        for(IResource res : task.getOutputs()) {
+            String path = res.getAbsPath();
+            if(path.endsWith("luac") || path.endsWith("scriptc") || path.endsWith("gui_scriptc") || path.endsWith("render_scriptc")) {
+                if (use_lua_source) {
                     project.addOutputFlags(path, Project.OutputFlags.UNCOMPRESSED);
                 }
+                else {
+                    project.addOutputFlags(path, Project.OutputFlags.ENCRYPTED);
+                }
             }
         }
 
         LuaSource.Builder srcBuilder = LuaSource.newBuilder();
         srcBuilder.setFilename(task.input(0).getPath());
 
-        if (needsLuaSource.contains(project.getPlatform()) || use_vanilla_lua) {
+        if (use_lua_source) {
             srcBuilder.setScript(ByteString.copyFrom(script.getBytes()));
+        } else if (needsVanillaLua32.contains(project.getPlatform())) {
+            byte[] bytecode = constructLuaBytecode(task, "luac-32", script);
```

Comment:
If we are not using luac-64, I don't think we should ship the executables?

Reply:
Removed luac-64